### PR TITLE
FIX: Ensuring both x and y attrs of LocationEvent are int

### PR DESCRIPTION
**Pull request by TarasKuzyo**
## PR Summary

Fixes issue #11496. Both `x` and `y` attributes of `matplotlib.backend_bases.LocationEvent` are casted to `int` in the constructor. 

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -1365,8 +1365,10 @@ def __init__(self, name, canvas, x, y, guiEvent=None):
         *x*, *y* in figure coords, 0,0 = bottom, left
         """
         Event.__init__(self, name, canvas, guiEvent=guiEvent)
-        self.x = x          # x position - pixels from left of canvas
-        self.y = y          # y position - pixels from right of canvas
+        # x position - pixels from left of canvas
+        self.x = int(x) if x is not None else 0
```
Comment (tacaswell) [inline lib/matplotlib/backend_bases.py]: We should probably leave the current behavior of x and y being allowed to be None.  I am not sure why that is allowed, but we should not change it!

**Comment by TarasKuzyo** [comment]
The only place, where `None` is explicitly passed to `x` and `y` attributes of `LocationEvent` is `FigureCanvasBase.enter_notify_event`:
https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backend_bases.py#L1858
On the other hand in `KeyEvent`'s constructor zeros are passed for `x` and `y`:
https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backend_bases.py#L1562

**Comment by NelleV** [comment]
Hi @TarasKuzyo 
Thanks for the contribution! This looks great. 
I think the only element missing is a test, which I know can be challenging sometimes with Matplotlib. Do you think you can try adding this? In this particular case, it should be fairly straight forward and I am happy to help you out if need be.

**Comment by TarasKuzyo** [comment]
Hi @NelleV, should this test simulate `button_press_event` on `Canvas` or just creating `MouseEvent` and checking appropriate attributes?

**Comment by NelleV** [comment]
@TarasKuzyo no, I would go for something simpler, that checks that in the object is created with a float, it is casted properly as an integer during the object initialization. It's more targeted and way easier to implement, but also a good way to check we don't accidentally remove this feature.

**Comment by NelleV** [comment]
Thanks @TarasKuzyo !

**Pull request merged**